crates/router/src/core/payment_methods/cards.rs

```diff
@@ -256,8 +254,20 @@ pub async fn add_card_hs(
     customer_id: String,
     merchant_account: &domain::MerchantAccount,
 ) -> errors::CustomResult<(api::PaymentMethodResponse, bool), errors::VaultError> {
-    let store_card_payload =
-        call_to_card_hs(state, &card, None, &customer_id, merchant_account).await?;
+    let payload = payment_methods::StoreLockerReq::LockerCard(payment_methods::StoreCardReq {
+        merchant_id: &merchant_account.merchant_id,
+        merchant_customer_id: customer_id.to_owned(),
+        card: payment_methods::Card {
+            card_number: card.card_number.to_owned(),
+            name_on_card: card.card_holder_name.to_owned(),
+            card_exp_month: card.card_exp_month.to_owned(),
+            card_exp_year: card.card_exp_year.to_owned(),
+            card_brand: None,
+            card_isin: None,
+            nick_name: card.nick_name.to_owned().map(masking::Secret::expose),
```

Comment:
How does this differ from `name_on_card`?  
I suggest you use `peek` from `peekinterface` instead of using `expose`

Reply:
`name_on_card` refers to the `card_holder_name` which is different from `nick_name`

Reply:
Why should we use `peek` instead of `expose`? I have a value which I'm mapping to expose, which returns the inner value.

If we were to use `peek`, the reference will be passed to the `peek` function, which would return a reference to the inner value, which would need to be dereferenced again

With use of `expose` we don't have to reference and de-reference it.

Reply:
- By using `peek()` you're just passing a reference and not the complete value and this increases the security by not giving complete ownership like what `expose()` does
- By using `expose()`, there exist a possibility to get the values changed since it is different value and not the reference to its parent